### PR TITLE
Allow numbers to set uvc for all arrows in quiver.set_UVC, fixes #16743

### LINKED ISSUE #16743: Breaking change in 3.2: quiver.set_UVC does not support single numbers any more
### Bug report

**Bug summary**

In prior versions of matplotlib, it was possible to set all UVC values to the same by calling `quiver.set_UVC` with single numbers for U and V. This raises an error in matplotlib 3.2.

**Code for reproduction**

```python
import matplotlib.pyplot as plt

q = plt.quiver(
    [0, 0, 1],
    [0, 1, 1],
    [0.5, 0.5, 0.5],
    [0, 0, 0],
)

q.set_UVC(0, 0.5)
plt.xlim(-1, 2)
plt.ylim(-1, 2)
plt.show()

```

**Actual outcome**

matplotlib 3.1 works as expected, matplotlib 3.2 raises:

```
    q.set_UVC(0, 0.5)
  File "/home/maxnoe/.local/anaconda3/lib/python3.7/site-packages/matplotlib/quiver.py", line 604, in set_UVC
    raise ValueError(f'Argument {name} has a size {var.size}'
ValueError: Argument U has a size 1 which does not match 3, the number of arrow positions
```

**Expected outcome**

No breaking changes between minor revisions.

### PR DESCRIPTION
**Pull request by MaxNoe**
## PR Checklist

- [X] Has Pytest style unit tests
- [X] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

Fix for #16743

### CODE REVIEW

**Comment by jklymak** [comment]
This is probably fine, but can you bisect and see why this was failing on 3.2 and what the PR was that made the failing change so we can be sure this wasn't purposeful?

**Comment by MaxNoe** [comment]
It is this: #14424

File: lib/matplotlib/quiver.py [hunk 1]
```diff
@@ -585,7 +585,7 @@ def set_UVC(self, U, V, C=None):
         if C is not None:
             C = ma.masked_invalid(C, copy=True).ravel()
         for name, var in zip(('U', 'V', 'C'), (U, V, C)):
-            if var is not None and var.size != self.N:
+            if var is not None and var.size != self.N and var.size != 1:
```
Comment (jklymak) [inline lib/matplotlib/quiver.py]: Maybe can this be rewritten as 
`if not (var is None or var.size == self.N or var.size==1):`  ?  I find three "nots" in a row starts to hurt my little head.

Comment (MaxNoe) [reply lib/matplotlib/quiver.py]: done

**Pull request merged**